src/com/company/Main.java

```diff
@@ -1,7 +1,154 @@
 package com.company;
 
+import java.util.Arrays;
+
 public class Main {
 
     public static void main(String[] args) {
+        common();
+
+        task4();
+
+        task5();
+
+        task6();
+
+        task7();
+
+        task8();
+
+        task9();
+
+    }
+
+    public static void common() {
+        int[] arr = generateRandomArray();
+
+        int sum = 0;
+        for (int i : arr) {
+            sum += i;
+        }
+        System.out.println("Сумма трат за месяц составила " + sum + " рублей");
+
+        int min = Integer.MAX_VALUE;
+        int max = Integer.MIN_VALUE;
+        for (int i : arr) {
+            if (i > max)
+                max = i;
+            if (i < min)
+                min = i;
+        }
+
+        System.out.println("Максимальная cумма трат за месяц составила " + max + " рублей");
+        System.out.println("Минимальная cумма трат за месяц составила " + min + " рублей");
+
+        float average = sum / (float) arr.length;
+
+        System.out.println("Средняя сумма трат за месяц составила " + average + " рублей");
+    }
+
+    public static void task4() {
+        char[] reverseFullName = "navI vonavI".toCharArray(); //мне было лень прописывать каждый символ :)
+
+        for (int i = 0; i < reverseFullName.length / 2; i++) {
+            final int index = reverseFullName.length - 1 - i;
+            final char tmp = reverseFullName[index];
+            reverseFullName[index] = reverseFullName[i];
+            reverseFullName[i] = tmp;
+        }
+        System.out.println(reverseFullName);
+    }
+
+    public static int[] generateRandomArray() {
+        java.util.Random random = new java.util.Random();
+        int[] arr = new int[30];
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = random.nextInt(100_000) + 100_000;
+        }
+        return arr;
+    }
+
+    public static void task5() {
+        System.out.println("\n\tЗадача 5\n");
+
+        int len = 3;
+        int[][] matrix = new int[len][len];
+        for (int i = 0; i < len * len; i++) {
+            int col = i % len;
+            int row = i / len;
+            if ((i % (len + 1)) == 0 || (i % (len - 1)) == 0)
+                matrix[col][row] = 1;
+        }
```

Comment:
более простой вариант:
```suggestion
         for (int i = 0; i < matrix.length; i++) {
           matrix[i][i] = 1;
           matrix[i][matrix.length - 1 - i] = 1;
       }
```